### PR TITLE
Don't split resumable code

### PR DESCRIPTION
**Pull request by vzarytovskii**
## Description

Decision tree splitting during optimisation had some low-ish thresholds, here we make sure we don't ever split if we see resumable code invocation (in this case we check for the application for Run/Delay, etc).

Fixes https://github.com/dotnet/fsharp/issues/16068

## Checklist

- [x] Test cases added
- [x] Release notes entry updated

### CODE REVIEW

**Comment by psfinaki** [comment]
I think I saw this warning suppressed in a couple of our projects (at least editor and editor.tests) - can the suppression be removed there now?

**Comment by vzarytovskii** [comment]
> I think I saw this warning suppressed in a couple of our projects (at least editor and editor.tests) - can the suppression be removed there now?

No, this fixes only specific case - when it was caused by function split. Others should be worked on 1 by 1, besides it's only in tests, and this warning (`FS3511`) by no means crucial, code still works, `RunDynamic` is still fast.

File: src/Compiler/Optimize/Optimizer.fs [hunk 1]
```diff
@@ -2329,6 +2329,13 @@ let rec IsDebugPipeRightExpr cenv expr =
         else false
     | _ -> false
 
+let inline IsStateMachineExpr g overallExpr =
+    //printfn "%s" (DebugPrint.showExpr overallExpr)
```
Comment (psfinaki) [inline src/Compiler/Optimize/Optimizer.fs:2333]:
```suggestion
```

File: tests/FSharp.Compiler.ComponentTests/Language/StateMachineTests.fs [hunk 2]
```diff
@@ -103,6 +103,61 @@ let compute () =
 for i in 1 .. 100 do
     compute().Wait ()
 """
+        |> withOptimize
+        |> compileExeAndRun
+        |> shouldSucceed
+
+    [<Fact>] // https://github.com/dotnet/fsharp/issues/16068
+    let ``Decision tree with 32+ binds with nested expression is not getting splitted and state machine is successfully statically compiles``() =
```
Comment (psfinaki) [inline tests/FSharp.Compiler.ComponentTests/Language/StateMachineTests.fs:111]:
```suggestion
    let ``Decision tree with 32+ binds with nested expression is not getting splitted and state machine is successfully statically compiled``() = 
```

**Review by psfinaki — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/Language/StateMachineTests.fs (hunk 2 above)
Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/Language/StateMachineTests.fs:111]: No, "whether it compiles or not"

**Review by T-Gro — APPROVED** [review]

**Pull request merged**